### PR TITLE
Remove BTM, an old Transaction Manager (#5487)

### PR DESCRIPTION
**Pull request by jkosternl**

### CODE REVIEW

File: core/src/test/java/nl/nn/adapterframework/receivers/ReceiverTest.java [hunk 1]
```diff
@@ -228,19 +224,10 @@ public ITransactionalStorage<Serializable> setupErrorStorage() {
 
 	public static Stream<Arguments> transactionManagers() {
 		return Stream.of(
-			Arguments.of(supplier(ReceiverTest::buildNarayanaTransactionManagerConfiguration)),
-			Arguments.of(supplier(ReceiverTest::buildBtmTransactionManagerConfiguration))
+			Arguments.of(supplier(ReceiverTest::buildNarayanaTransactionManagerConfiguration))
```
Comment (tnleeuw) [inline core/src/test/java/nl/nn/adapterframework/receivers/ReceiverTest.java:227]: Since there's still a chance that we might add new transaction manager implementations in future I think we should keep these tests parameterised.

File: test/src/test/testtool/TransactionHandling/scenario02.properties [hunk 2]
```diff
@@ -17,6 +17,6 @@ ignoreContentBetweenKeys.interruptedException.key2=</message>
 replaceRegularExpressionKeys.msgWas.key1=<message>null</message>
 replaceRegularExpressionKeys.msgWas.key2=<message>REPLACED</message>
 
-#TOMCATBTM/TOMCAT6
+#TOMCAT6
```
Comment (tnleeuw) [inline test/src/test/testtool/TransactionHandling/scenario02.properties:20]: Still `TOMCAT6`? Shall we update this comment to `TOMCAT9` for now? 😉 

(And soon perhaps 10 or 11)

**Review by tnleeuw — COMMENTED** [review]
This PR is going to break the CI after it is merged, because the CI still tries to run tests with BTM.

So we'll need to set up separate CI for 7.9 and master pretty soon.

File: test/src/test/testtool/TransactionHandling/scenario02.properties (hunk 2 above)
Comment (jkosternl) [reply test/src/test/testtool/TransactionHandling/scenario02.properties:20]: A version number does not make sense in this property file. Removing it.

File: core/src/main/java/nl/nn/adapterframework/lifecycle/IbisApplicationInitializer.java [hunk 3]
```diff
@@ -114,21 +113,6 @@ public WebApplicationContext initWebApplicationContext(ServletContext servletCon
 		}
 	}
 
-	//TODO: remove this in 8.0
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/lifecycle/IbisApplicationInitializer.java:117]: Perhaps a bit tricky to remove this in the PR for removing BTM although I agree that this code should be removed.
Perhaps better to do that in another PR, there's a lot more WAS related code that needs to be removed.

**Review by tnleeuw — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/lifecycle/IbisApplicationInitializer.java (hunk 3 above)
Comment (jkosternl) [reply core/src/main/java/nl/nn/adapterframework/lifecycle/IbisApplicationInitializer.java:117]: I agree. Done! #5525

**Review by tnleeuw — APPROVED** [review]
In principle this PR looks OK to me now.

**Review by nielsm5 — APPROVED** [review]
